staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness.go

```diff
@@ -111,7 +111,16 @@ func WithPriorityAndFairness(
 			httplog.AddKeyValue(ctx, "apf_pl", truncateLogField(pl.Name))
 			httplog.AddKeyValue(ctx, "apf_fs", truncateLogField(fs.Name))
 			httplog.AddKeyValue(ctx, "apf_fd", truncateLogField(flowDistinguisher))
-			return workEstimator(r, fs.Name, pl.Name)
+		}
+		// estimateWork is called, if at all, after noteFn
+		estimateWork := func() flowcontrolrequest.WorkEstimate {
+			if classification == nil {
+				// workEstimator is being invoked before classification of
+				// the request has completed, we should never be here though.
+				panic(fmt.Sprintf("workEstimator is being invoked before classification of the request has completed, verb: %s, URI: %s", r.Method, r.RequestURI))
```

Comment:
Please don't panic.
Let's log an error, return an error or sth.

I really prefer to have bad estimate or fail an individual request than to fire a panic.

Reply:
yes, i agree with that. replaced it with logging an error and falling back to empty string